src/rust/src/unified_keys_ffi.rs

```diff
@@ -129,3 +132,41 @@ pub extern "C" fn unified_full_viewing_key_from_components(
         }
     }
 }
+
+/// Selects a preferred ovk given input notes from multiple different pools
+/// from the same unified full viewing key.
+#[no_mangle]
+pub extern "C" fn unified_full_viewing_key_preferred_ovk(key: *const Ufvk, out: *mut [u8; 32]) {
+    todo!()
+}
+
+/// Derives an outgoing viewing key from a transparent full viewing key.
+#[no_mangle]
+pub extern "C" fn unified_full_viewing_key_transparent_ovk(
+    key: *const Ufvk,
+    external: bool,
+    out: *mut [u8; 32],
+) -> bool {
+    let key = unsafe { key.as_ref() }.expect("Unified full viewing key pointer may not be null.");
+    let out = unsafe { &mut *out };
+
+    for r in &key.items() {
+        if let Fvk::P2pkh(data) = r {
+            let c: [u8; 32] = data[0..32].try_into().unwrap();
+            let c: ChainCode = c.into();
+            let pk =
+                secp256k1::PublicKey::from_slice(&data[32..]).expect("pk is already validated.");
+            let fvk: transparent::FullViewingKey = (c, pk).into();
+
+            if external {
+                *out = fvk.external_ovk().as_bytes();
+            } else {
+                *out = fvk.internal_ovk().as_bytes();
+            }
+
+            return true;
+        }
+    }
+
+    false
```

Comment:
Use `find_ovk` defined at https://github.com/zcash/librustzcash/pull/480/files#diff-3b307a8fabc1b1ac923d7b23188c2fd995a416495c210a148a9914e7728d503aR86

```suggestion
    if let Some(ovk) = find_ovk(key, external) {
        *out = ovk.as_bytes();
        true
    } else {
        *out = [0; 32];
        false
    }
```
(I'm not sure if `*out = [0; 32];` is correct. In any case, zeroing the output array on error is probably a good idea.)

Reply:
Resolving because the implementation has changed significantly.